spark/v3.2/spark/src/main/java/org/apache/iceberg/spark/Spark3Util.java

```diff
@@ -448,6 +451,19 @@ public static boolean extensionsEnabled(SparkSession spark) {
     return extensions.contains("IcebergSparkSessionExtensions");
   }
 
+  public static String toIcebergField(String name) {
+    Preconditions.checkArgument(!name.isEmpty(), "Invalid column name: (empty)");
+    if (!isCaseSensitiveEnabled()) {
```

Comment:
This isn't the right way to make sort order case insensitive. If the table was created with an upper case name, this would no longer work.

I think the right way to fix this problem is to set the `caseSensitive` config property in the `SortOrderBuilder`. That is currently defaulted to `true`. I think you need to expose a way to set that and call it appropriately.